### ISSUE TITLE
PLI adjustment

### ISSUE THREAD

**Issue by ErfinderDesRades** [trevorsandy/lpub3d#685]
During the last weeks I have been dealing a lot with LPub3D. I learned a lot and the cases where I get the expected results are getting more and more. But when it comes to adjusting the objects on a multi step page, there are stil a lot of situations where I experience results I didn't expect. As I'm still far away from being an expert in LPub3D I'm not sure whether these behaviour are are to considere as issues or are caused by mistakes on my side.

When it comes to aligning the PLI in relation to the assembly there are a lot of preset adjustments, which do not show the expected result.

The following adjustments work as expected:

![image](https://user-images.githubusercontent.com/77791504/217071762-53528b5a-d5f6-4c20-ba65-a329da6c632a.png)

![image](https://user-images.githubusercontent.com/77791504/217072111-dbb80dd3-209a-4469-97fa-e408627f9fe7.png)

![image](https://user-images.githubusercontent.com/77791504/217072195-e3a205c9-7078-4004-92eb-121e980441e5.png)

![image](https://user-images.githubusercontent.com/77791504/217072347-92739777-ea9a-4858-9d39-df2e199ab9d6.png)

![image](https://user-images.githubusercontent.com/77791504/217072385-e5d2f130-2793-4389-aabb-94fa9546b4b8.png)

Interesting:
When you go back from here to, for example Top:Center, the adjustment of the PLI jumps back to Top:Left.

![image](https://user-images.githubusercontent.com/77791504/217072427-3610821d-bb0d-4df7-96a8-45045b803d3b.png)

![image](https://user-images.githubusercontent.com/77791504/217072476-c716a260-5acb-4b55-8239-227b8a1eef8b.png)

When choosing the other preset positions, the PLI jumps back to Top:Left or remains at the last position.

Also interesting:

- no matter which position you choose, when opening the menu again the Left:Top button is bold
- when changing the position in the menu, the new code line containing the new adjustment will be written above the old one. So even if the preset psoition is one of the working ones, it will not work before deleting the "old" line.

It would really be highly appreciated, if you could clarify these "issues". Or maybe tell me, what are my mistakes.

To be sure I attached the model file code, althought I haven't changed any formatting, execpt the scale of the PLIs in order to make changes more obvious.

0 Example Pyramid for Demonstration of LDRAW Library
0 Name: pyramid.ldr
0 Author: James Jessiman
0 !LPUB MULTI_STEP BEGIN
0 !LPUB PLI MODEL_SCALE  0.4000
1 1 -40 -24 60 1 0 0 0 1 0 0 0 1 3001.dat
1 1 40 -24 60 1 0 0 0 1 0 0 0 1 3001.dat
1 1 60 -24 0 0 0 1 0 1 0 -1 0 0 3001.dat
1 1 40 -24 -60 1 0 0 0 1 0 0 0 1 3001.dat
1 1 -40 -24 -60 1 0 0 0 1 0 0 0 1 3001.dat
1 1 -60 -24 0 0 0 1 0 1 0 -1 0 0 3001.dat
0 STEP
1 4 -20 -48 40 1 0 0 0 1 0 0 0 1 3001.dat
1 4 40 -48 20 0 0 1 0 1 0 -1 0 0 3001.dat
1 4 20 -48 -40 1 0 0 0 1 0 0 0 1 3001.dat
1 4 -40 -48 -20 0 0 1 0 1 0 -1 0 0 3001.dat
0 STEP
1 14 0 -72 20 1 0 0 0 1 0 0 0 1 3001.dat
1 14 0 -72 -20 1 0 0 0 1 0 0 0 1 3001.dat
0 STEP
0 !LPUB MULTI_STEP END
1 0 0 -96 0 1 0 0 0 1 0 0 0 1 3003.dat
0 STEP
0 STEP
0 // These 6 command lines were auto-generated for highlight current step.
0 // These lines are not saved and should not be modified, but they can be replaced.
0 // Remove comment lines and replace MODEL with DISPLAY_MODEL to override this behaviour.
0 STEP
0 // These 6 command lines were auto-generated for highlight current step.
0 // These lines are not saved and should not be modified, but they can be replaced.
0 // Remove comment lines and replace MODEL with DISPLAY_MODEL to override this behaviour.
0 STEP

_Environment
Version of LPub3D - 2.4.6.0.3111_20230111
Windows 11, Processor 12th Gen Intel(R) Core(TM) i7-12700H 2.70 GHz
Architecture 64-Bit, 32 GB Ram
Configuration
LPub3D installation path
Windows: C:\Program Files\LPub3D_246
Application data path
Until now I couldn't find any workaround_

Best regards and thanks a lot in advance.

**Comment by trevorsandy** [comment]
Thank you for reporting this behaviour.

I'll take a look.

Cheers,

**Comment by trevorsandy** [comment]
This behaviour has been corrected and will be available in the next release build

> When you go back from here to, for example Top:Center, the adjustment of the PLI jumps back to Top:Left.

Corrected.

> when changing the position in the menu, the new code line containing the new adjustment will be written above the old one. So even if the preset position is one of the working ones, it will not work before deleting the "old" line.

I was not able to reproduce this behaviour using the sample model file you provided after implementing the correction.

> It would really be highly appreciated, if you could clarify these "issues". Or maybe tell me, what are my mistakes.

The unexpected behaviour was caused by the `LOCAL` meta command flag not being properly persisted when the multi-step page is sized for display. If you remove LOCAL from the meta command (you will have to place a 'restore' meta command in any subsequent page STEP that you do not wish to inherit the 'local' placement), this behaviour will not present.

For example, set this meta command:

`0 !LPUB MULTI_STEP PLI PLACEMENT LOCAL RIGHT CENTER ASSEM OUTSIDE`

To:

`0 !LPUB MULTI_STEP PLI PLACEMENT RIGHT CENTER ASSEM OUTSIDE`

In route to correct this behaviour I've discovered other anomalies with the placement routine so I'll also correct these items before the next release build. 

Cheers,
